### ISSUE TITLE
[Bug] translatable models not showing correct locale

### ISSUE THREAD

**Issue by kandrejevs** [Laravel-Backpack/CRUD#4307]
# Bug report

### What I did

Upgraded to latest backpack version.
Translatable models language switcher stopped working, always defaults to framework default language.

### What I expected to happen

Locale switcher should load correct language for model

### What happened

Changing locale always returns default locale, but editing and saving it stores translation correctly

### What I've already tried to fix it

Downgraded backpack/crud to 5.0.10 version fixes issue, so that is something introduced in 5.0.11.
I have also replicated same behavior on clean laravel installation.
I believe that root cause of this problem can be found here https://github.com/Laravel-Backpack/CRUD/commit/4bc41e9fc605f87861a1ca8170adac505189edab

### Is it a bug in the latest version of Backpack?

After I run ```composer update backpack/crud``` the bug... is it still there?
It appeared after update

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:
```
### PHP VERSION:
PHP 8.1.3 (cli) (built: Feb 21 2022 14:48:58) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.3, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.3, Copyright (c), by Zend Technologies
    with Xdebug v3.1.2, Copyright (c) 2002-2021, by Derick Rethans

### LARAVEL VERSION:
v9.6.0@47940a1a8774b96696ed57e6736ccea4a880c057

### BACKPACK VERSION:
5.0.13@d78f4263f5eef1e7761ba2d55d81959f4de559b6
```

**Comment by remipou** [comment]
I got the same issue, fixed when downgrading from 5.0.13 to 5.0.10

**Comment by remipou** [comment]
I got the same issue, fixed when downgrading from 5.0.13 to 5.0.10

**Comment by pxpm** [comment]
Hello @kandrejevs and @remipou ! Thanks for the report.

Totally right, sorry for the bad experience. This was an attempt to fix other problem but I forgot about the translatable magic methods. 

I've submitted #4313 that should fix this, so I am going to close. If you happen to test the solution I provided in #4313 let us know in PR that it worked for you.

Cheers
